lib/sles4sap_publiccloud.pm

```diff
@@ -608,4 +610,63 @@ sub delete_network_peering {
     }
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns 1 if hana database is online otherwise 0 offline.
+
+=over 1
+
+=item B<RESULT> - default "-1" representing empty value.
+
+=back
+=cut
+
+sub get_hana_database_info {
+    my ($self) = @_;
+    my $result = -1;
+    $self->is_hana_online();
+    my $output_cmd = $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'", runas => get_required_var("SAP_SIDADM"));
+    my $hana_database_online = '"HDB","HDB-00","YES","ACTIVE","","","DEFAULT","?"';
+
+    $result = ($output_cmd =~ /$hana_database_online/ ? 1 : 0);
+    record_info("Hana database status", $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'"));
```

Comment:
Same comment about the password and the instance number as above.

Reply:
refactored